### PR TITLE
Move wheel support into a Python library, create a `starforge` command

### LINKED ISSUE #38: Select a random port for forwarding SSH when running under QEMU
And create an ephemeral ssh config directory and ssh config. This is roughly what Vagrant does.

### PR DESCRIPTION
**Pull request by natefoo**
(Yes I realize I sorta argued against this level of complexity in #15, sorry!)

TODO: expand to incorporate regular dep building with @erasche's "V2" `build.py` and maybe even the shell-based recipe handling.

Also, fixes #38

### CODE REVIEW

File: starforge/config/default.yml [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+    starforge/base32-wheel:
+        plat_name: linux_i686
+    starforge/centos-wheel:6:
+        pkgtool: yum
+        plat_name: linux_x86_64_rhel_6
+    starforge/centos-wheel:7:
+        pkgtool: yum
+        plat_name: linux_x86_64_rhel_7
+    starforge/centos32-wheel:6:
+        pkgtool: yum
```
Comment (bgruening) [inline starforge/config/default.yml:69]: Uh I like this!

**Comment by bgruening** [comment]
Given the changed aim of starforge let's concentrate on V2 from now on. It will be a lot of work to support both.
:+1:  This is awesome!

File: starforge/execution/docker.py [hunk 2]
```diff
@@ -0,0 +1,51 @@
+"""
+"""
+from __future__ import absolute_import
+
+from subprocess import check_call
+try:
+    from subprocess import check_output
+except ImportError:
+    from ..util import check_output
```
Comment (hexylena) [inline starforge/execution/docker.py:9]: ... do we really need this? Do we _really_ need to support py2.6?

**Comment by hexylena** [comment]
This is a lot of new code, wow. Docker-build has really moved on.

File: starforge/execution/docker.py (hunk 2 above)
Comment (natefoo) [inline starforge/execution/docker.py:9]: Both squeeze and EL6 use Python 2.6.

Probably not a ton of squeeze boxes out there anymore, but plenty of EL6es.

**Pull request merged**